tequilapi/endpoints/identities.go

```diff
@@ -226,14 +299,25 @@ func validateCreationRequest(createReq *identityCreationDto) (errors *validation
 	return
 }
 
+func validatePayoutInfoRequest(req *payoutInfoDto) (errors *validation.FieldErrorMap) {
+	errors = validation.NewErrorMap()
+	if req.EthAddress == nil {
+		errors.ForField("eth_address").AddError("required", "Field is required")
+	}
+	// TODO: implement validation of eth address
+	return
+}
+
 //AddRoutesForIdentities creates /identities endpoint on tequilapi service
 func AddRoutesForIdentities(
 	router *httprouter.Router,
 	idm identity.Manager,
 	signerFactory identity.SignerFactory,
+	payoutInfoRegistry PayoutInfoRegistry,
 ) {
-	idmEnd := NewIdentitiesEndpoint(idm, signerFactory)
+	idmEnd := NewIdentitiesEndpoint(idm, signerFactory, payoutInfoRegistry)
 	router.GET("/identities", idmEnd.List)
 	router.POST("/identities", idmEnd.Create)
 	router.PUT("/identities/:id/unlock", idmEnd.Unlock)
+	router.POST("/identities/:id/payout", idmEnd.CreatePayoutInfo)
```

Comment:
It should be `POST` + `UpdatePayoutInfo`, because `POST` usually stands for adding new resource in REST terminology

Reply:
👍

Reply:
This endpoint is for creating, not updating.

Reply:
POST vs PUT is not about Creating/updating. Its about adding and replacing. Are you adding to a list of payouts or just creating a payout info? If I repeat PUT with several times with the same info - do I create a list of payouts or simply update existing one? Thats the semantics of PUT and its perfectly fine here.

Reply:
Feature should act so, that user can always come back and update his payout information

Reply:
Agree. Change to PUT